doc-surrealdb_versioned_docs/version-1.x/tutorials/define-a-schema.mdx

```diff
@@ -0,0 +1,230 @@
+---
+sidebar_position: 6
+sidebar_label: Define a Schema in SurrealDB
+title: Define a Schema in SurrealDB | Tutorials
+description: In this tutorial, you will learn how to define a schema in SurrealDB, what using either schema type means for data retrieval, and how to use what you need as your product grows.
+---
+
+# Define a Schema in SurrealDB
+
+
+When starting a new database project, there are a couple of early decisions to be made, such as creating tables and defining the records that will be in these tables and also the datatypes of the records, and how the tables you describe relate to each other which includes data sharing or manipulation.
+
+While this might look like many decisions, you are Defining a Schema.
+
+A Schema defines the structure and organisation of data. It dictates how data is stored, organised, and manipulated. Schemas can specify tables, fields (columns), data types, constraints, and relationships between tables.
+
+In summary, a schema is the primary way to ensure your data acts as expected.
+
+There are two main types of schemas:
+
+1. **Schemafull (Structured)**
+2. **Schemaless (Unstructured)**
+
+When starting a new project in SurrealDB, you can define your Schema using either method depending on your application's requirements. Since SurrealDB is a multi-model database, you can have both in the same project, depending on your needs.
+
+In this tutorial, you will learn how to define a schema in SurrealDB, what using either schema type means for data retrieval, and how to use what you need as your product grows.
+
+## Prerequisites
+
+Before you start, this guide assumes the following:
+
+- Basic knowledge of databases and data modelling.
+- SurrealDB is installed on your machine. You can download and install SurrealDB from the [installation page](/docs/surrealdb/installation/).
+- [A Command line interface (CLI)](/docs/surrealdb/cli/sql) for interacting with SurrealDB or a [Surrealist sandbox](/docs/surrealist/#connecting-to-the-sandbox).
+
+## Schemafull (Structured) Databases
+
+A schemafull database requires the upfront definition of the structure of your data, including collections (tables) and fields (columns). This approach enforces consistency and integrity, making it suitable for applications with well-defined data models.
+
+In surrealDB, the Schemafull approach is usually associated with the [Define statement](/docs/surrealdb/surrealql/statements/define/), which is used to specify instructions to the Schema, such as authentication access and behaviour, global parameters, table configurations, table events, analyzers, and indexes. You can set a schemafull table in the following steps:
+
+### Creating a schemafull table
+
+1. Define a Table: To start a schemafull table, specifically use the Define Table statement.
+
+```surql
+-- Create a schemafull user table.
+DEFINE TABLE user SCHEMAFULL;
+```
+
+2. Define Fields:
```

Comment:
Here I think it would be nice to mention why the DEFINE FIELD statements are used, that at this point you can't set any fields at all until you use them.

"Now that the table is schemafull, no fields can be set unless first defined through a DEFINE FIELD statement."